superset-frontend/packages/superset-ui-core/src/connection/callApi/parseResponse.ts

```diff
@@ -56,7 +56,10 @@ export default async function parseResponse<T extends ParseMethod = 'json'>(
       // `json-bigint` could not handle floats well, see sidorares/json-bigint#62
       // TODO: clean up after json-bigint>1.0.1 is released
       json: cloneDeepWith(json, (value: any) =>
-        value?.isInteger?.() === false ? Number(value) : undefined,
+        value?.isInteger?.() === false ||
```

Comment:
@justinpark how is there any merit to casting this to a [Decimal](https://mikemcl.github.io/decimal.js/) as opposed to a `Number` or are these types typically not used by the Superset frontend?

Reply:
Good point. I also concerned about the data inaccuracy issue since this will convert all float numbers in Number(even Decimal) `234020423042304302.402` vs `234020423042304300 (=Number(234020423042304302.402))`

I reverted this json-bigint modification and made the extra date value conversion logic in SqlLab instead.